app/Jobs/DeleteWikiFinalizeJob.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace App\Jobs;
+use App\Wiki;
+use App\WikiSetting;
+use App\WikiManager;
+use App\QueryserviceNamespace;
+use App\WikiDb;
+use Traversable;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Support\Facades\Storage;
+use App\Http\Controllers\WikiLogoController;
+use Illuminate\Contracts\Filesystem\Cloud;
+use Illuminate\Contracts\Queue\ShouldBeUnique;
+
+class DeleteWikiFinalizeJob extends Job implements ShouldBeUnique
+{
+    private $wikiId;
+
+    /**
+     * @return void
+     */
+    public function __construct( $wikiId )
+    {
+        $this->wikiId = $wikiId;
+    }
+
+    /**
+     * The unique ID of the job.
+     *
+     * @return string
+     */
+    public function uniqueId()
+    {
+        return $this->wikiId;
+    }
+
+    /**
+     * @return void
+     */
+    public function handle()
+    {
+        $wiki = Wiki::withTrashed()->where('id', $this->wikiId )->first();
+
+        if( !$wiki->deleted_at ) {
+            $this->fail(new \RuntimeException("Wiki {$this->wikiId} is not deleted, but job got dispatched."));
+            return;
+        }
+
+        if( !$wiki ) {
+            $this->fail(new \RuntimeException("Wiki not found for {$this->wikiId}"));
+            return;
+        }
+
+        $qsNamespace = $wiki->wikiQueryserviceNamespace()->first();
+        $wikiDB = $wiki->wikiDb()->first();
+
+
+        if( $wikiDB || $qsNamespace ) {
+            $this->fail(new \RuntimeException("There are still resources allocated."));
+            return;
+        }
+        
+        // if no settings there shouldn't be a need
+        $needToDeleteLogosDir = $wiki->settings()->whereIn('name', array(WikiSetting::wgFavicon, WikiSetting::wgLogo))->count() > 0;
```

Comment:
This should probably change to needsToDeleteSiteDir or something like that.

`sites/'.$siteDir`.

There is probably nota  directory unless people upload a logo right now,
so I imagine the logic would be something like:
 - If site dir exists, delete it, if not don't worry.